packages/frontend/app/components/ui/LiquidityDialog.tsx

```diff
@@ -0,0 +1,62 @@
+import { Dialog } from '@headlessui/react'
+import { Form } from '@remix-run/react'
+import { XIcon } from '~/components/icons'
+import { Button, Input } from '~/components/ui'
+
+type LiquidityDialogProps = {
+  title: string
+  onClose: () => void
+  type: 'Add' | 'Withdraw'
+}
+
+export const LiquidityDialog = ({
+  title,
+  onClose,
+  type
+}: LiquidityDialogProps) => {
+  return (
+    <Dialog as='div' className='relative z-10' onClose={onClose} open={true}>
+      <div className='fixed inset-0 bg-tealish/30 bg-opacity-75 transition-opacity' />
+      <div className='fixed inset-0 z-10 overflow-y-auto'>
+        <div className='flex min-h-full items-center justify-center p-4 text-center'>
+          <Dialog.Panel className='relative transform overflow-hidden rounded-lg max-w-full transition-all bg-white px-4 pb-4 pt-5 text-left shadow-xl w-full sm:max-w-lg'>
+            <div className='absolute right-0 top-0 pr-4 pt-4'>
+              <button
+                type='button'
+                className='text-gray-400 hover:text-gray-500 focus:outline-none'
+                onClick={onClose}
+              >
+                <span className='sr-only'>Close</span>
+                <XIcon className='h-8 w-8' aria-hidden='true' />
+              </button>
+            </div>
+            <div>
+              <Dialog.Title
+                as='h3'
+                className='font-semibold leading-6 text-lg text-center'
+              >
+                {title}
+              </Dialog.Title>
+              <div className='mt-2'>
+                <Form method='post' replace preventScrollReset>
```

Comment:
Based on the UI flashing, I think this will trigger the `isSubmitting` on the parent `peers.$peerId.tsx` page. Is there a way to decouple this? If no easy way, NABD at all, just something I noticed

Reply:
Not really direct way to decouple them since the dialog is rendered as a child for `peers.$peerId.tsx`.